http-server/src/main/java/org/triplea/server/remote/actions/SessionTracker.java

```diff
@@ -0,0 +1,63 @@
+package org.triplea.server.remote.actions;
+
+import java.net.InetAddress;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.websocket.Session;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+@Builder
+@Slf4j
+class SessionTracker {
+  private final Map<InetAddress, Map<String, Session>> listeningSessions = new HashMap<>();
+
+  @Nonnull private final Function<Session, InetAddress> ipAddressExtractor;
+
+  void addSession(final Session session) {
+    final InetAddress ip = ipAddressExtractor.apply(session);
+    log.info("Adding websocket session, IP: {}, session id: {}", ip, session.getId());
+
+    if (!listeningSessions.containsKey(ip)) {
+      listeningSessions.put(ip, new HashMap<>());
+    }
+    final Map<String, Session> sessionsById = listeningSessions.get(ip);
+    sessionsById.put(session.getId(), session);
```

Comment:
```java
listeningSessions.computeIfAbsent​(ip, k -> new HashMap<>()).put(session.getId(), session)
```

Reply:
TBH 'computeIfAbsent' is not my favorite API. It's a reasonable change though, updated: ccaa761a1